### PR TITLE
Adding TryResolveAsync helper method

### PR DESCRIPTION
**Pull request by glennblock**
This method is a helper for handling bubbling errors from the resolve lambda. It will invoke an async lambda provided by the user wrapping with a try/catch. If an Exception occurs it will grab the message and add it to the Errors collection of the context. The Try method also allows providing a custom error handler for more complex logic.

### CODE REVIEW

**Comment by glennblock** [comment]
@joemcbride thoughts?

File: src/GraphQL/Types/ResolveFieldContext.cs [hunk 1]
```diff
@@ -106,6 +108,27 @@ public bool HasArgument(string argumentName)
         {
             return Arguments?.ContainsKey(argumentName) ?? false;
         }
+
+        public async Task<object> TryAsyncResolve(Func<ResolveFieldContext<TSource>, Task<object>> resolve, Func<ExecutionErrors, Task<object>> error = null)
+        {
+            try
+            {
+                var result = await resolve(this);
+                return Task.FromResult(result);
```
Comment (joemcbride) [inline src/GraphQL/Types/ResolveFieldContext.cs]: I think can just return `result` here.

Comment (glennblock) [reply src/GraphQL/Types/ResolveFieldContext.cs]: Thanks use to not wrap it but it was not working right. I wlll test again.

**Comment by glennblock** [comment]
@joemcbride verified that worked. Updated the code/tests. Thanks!

File: src/GraphQL/Types/ResolveFieldContext.cs [hunk 2]
```diff
@@ -106,6 +108,26 @@ public bool HasArgument(string argumentName)
...
+        {
+            try
+            {
+                return await resolve(this);
+            }
+            catch (Exception ex)
+            {
+                if (error == null)
+                {
+                    Errors.Add(new ExecutionError(ex.Message));
```
Comment (pekkah) [inline src/GraphQL/Types/ResolveFieldContext.cs]: Do we lose context if only exception message is added? Should it contain the whole ex.ToString()?

Comment (pekkah) [reply src/GraphQL/Types/ResolveFieldContext.cs]: Especially in case of AggregateException etc..

File: global.json [hunk 3]
```diff
@@ -1,5 +0,0 @@
-{
-  "sdk": {
-    "version": "2.0.0"
-  }
-}
```
Comment (joemcbride) [inline global.json]: Would like to keep this file as it signals to AppVeyor what SDK to use.

File: src/GraphQL/Types/ResolveFieldContext.cs (hunk 2 above)
Comment (joemcbride) [reply src/GraphQL/Types/ResolveFieldContext.cs]: Could perhaps add the full exception as the inner exception to keep the simpler top level message.

File: global.json (hunk 3 above)
Comment (glennblock) [reply global.json]: OK, will add it back

File: src/GraphQL/Types/ResolveFieldContext.cs (hunk 2 above)
Comment (glennblock) [reply src/GraphQL/Types/ResolveFieldContext.cs]: @joemcbride the point of this method is to allow nicer messages. In my case for example I am throwing a contextual message to the user. I didn't include the stack trace / inner exception for that reason. I guess one option is to have a switch that lets you decide if you want the inner exception.

Comment (joemcbride) [reply src/GraphQL/Types/ResolveFieldContext.cs]: The inner exception will never been seen in the response unless you have `ExposeExceptions` set on the `ExecutionOptions`.  I'm fine with either way.

**Comment by glennblock** [comment]
@pekkah @joemcbride adding setting the inner exception now, and added back the global.json.

**Comment by glennblock** [comment]
👍

**Pull request merged**